src/main/java/org/jenkinsci/plugins/ewm/strategies/DiskAllocationStrategy.java

```diff
@@ -0,0 +1,78 @@
+package org.jenkinsci.plugins.ewm.strategies;
+
+import org.jenkinsci.plugins.ewm.definitions.Disk;
+import org.jenkinsci.plugins.ewm.definitions.DiskPool;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+/**
+ * Abstract class for defining a disk allocation strategy.
+ * Each disk allocation strategy should extend this class and provide its own implementation of disk allocation.
+ *
+ * @author Alexandru Somai
+ */
+public abstract class DiskAllocationStrategy {
+
+    private final String diskPoolId;
+    private final List<DiskPool> diskPools;
+
+    protected DiskAllocationStrategy(@Nonnull String diskPoolId, @Nonnull List<DiskPool> diskPools) {
+        this.diskPoolId = diskPoolId;
+        this.diskPools = diskPools;
+    }
+
+    /**
+     * Allocates a disk from the given disk pool.
+     *
+     * @return the allocated disk pool
+     * @throws RuntimeException if the disk pool doesn't have defined any {@link Disk} entries
+     * @see DiskAllocationStrategy#findDiskPool()
+     * @see DiskAllocationStrategy#allocateDisk(List) ()
+     */
+    @Nonnull
+    public Disk allocateDisk() {
+        DiskPool diskPool = findDiskPool();
+
+        List<Disk> disks = diskPool.getDisks();
+        if (disks.isEmpty()) {
+            String message = String.format("No Disks were defined in the global config for Disk Pool '%s'", diskPoolId);
+            throw new RuntimeException(message);
```

Comment:
@alexsomai, it is the `AbortException` I was thinking about on Monday at the meeting.